src/dataset-builder/DomainCriteriaSelector.ts

```diff
@@ -60,19 +65,44 @@ export const saveSelected =
 
 export const DomainCriteriaSelector = (props: DomainCriteriaSelectorProps) => {
   const [rootConcepts, loadRootConcepts] = useLoadedData<GetConceptsResponse>();
+  const [hierarchyConcepts, loadHierarchyConcepts] = useLoadedData<GetConceptsHierarchyMapResponse>();
   const { state, onStateChange, datasetId, getNextCriteriaIndex } = props;
+  const [hierarchy, setHierarchy] = useState(new Map<number, Concept[]>());
+  const [isHierarchyLoaded, setIsHierarchyLoaded] = useState(false);
   useOnMount(() => {
-    void loadRootConcepts(() => DataRepo().dataset(datasetId).getConcepts(state.domainOption.root));
+    const openedConcept = state.openedConcept;
+    if (openedConcept) {
+      void loadHierarchyConcepts(() => DataRepo().dataset(datasetId).getConceptsHierarchy(openedConcept));
```

Comment:
I think there's a way to write this so that you only have one state that represents both `if` branches, by using `setHierarchy` for both cases. If so it'd be nice to eliminate the additional state objects.